src/query/expression/src/types.rs

```diff
@@ -344,6 +346,54 @@ pub trait ValueType: Debug + Clone + PartialEq + Sized + 'static {
     fn column_memory_size(col: &Self::Column) -> usize {
         Self::column_len(col) * std::mem::size_of::<Self::Scalar>()
     }
+
+    // Compare two scalars and return the Ordering between them.
+    #[inline(always)]
+    fn compare(_: Self::ScalarRef<'_>, _: Self::ScalarRef<'_>) -> Ordering {
+        unreachable!()
+    }
+
+    // Return the comparison function for the given select operation, some data types not support comparison.
+    #[inline(always)]
+    fn compare_operation(_op: &SelectOp) -> fn(Self::ScalarRef<'_>, Self::ScalarRef<'_>) -> bool {
+        unreachable!()
+    }
+
+    // Equal comparison between two scalars, some data types not support it.
+    #[inline(always)]
+    fn equal(_left: Self::ScalarRef<'_>, _right: Self::ScalarRef<'_>) -> bool {
```

Comment:
use default trait implementation instead of implementing them in a derived struct.

```
fn equal(_left: Self::ScalarRef<'_>, _right: Self::ScalarRef<'_>) -> bool {
   Self::compare(..).is_equal()
}
```